velox/docs/functions/presto/conversion.rst

```diff
@@ -0,0 +1,266 @@
+Conversion Functions
+====================
+
+During expression evaluations, Velox doesn't implicitly convert expression
+arguments to the correct types. If such a conversion is necessary, two provided
+conversion functions can be used explicitly cast values to a particular type.
+
+Conversion Functions
+--------------------
+
+.. function:: cast(value AS type) -> type
+
+    Explicitly cast a value as a type. This can be used to cast a varchar to a
+    numeric value type and vice versa.
+
+.. function:: try_cast(value AS type) -> type
+
+    Like :func:`cast`, but returns null if the cast fails. ``try_cast(x AS type)``
+    is different from ``try(cast(x AS type))`` in that ``try_cast`` only suppresses
+    errors happening during the casting itself but not those during the evaluation
+    of its argument. For example, ``try_cast(x / 0 as double)`` throws a divide-by-0
+    error, while ``try(cast(x / 0 as double))`` returns a NULL.
+
+Supported Conversions
+---------------------
+
+The supported conversions are listed below, with from-types given at rows and to-types given at columns. Conversions of ARRAY, MAP, and ROW types
+are supported if the conversion of their element types are supported. In addition,
+supported conversions to/from JSON are listed in :doc:`json`.
+
+.. list-table::
+   :widths: 25 25 25 25 25 25 25 25 25 25 25 25
+   :header-rows: 1
+
+   * -
+     - tinyint
+     - smallint
+     - integer
+     - bigint
+     - boolean
+     - real
+     - double
+     - varchar
+     - timestamp
+     - date
+     - decimal
+   * - tinyint
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     - Y
+   * - smallint
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     - Y
+   * - integer
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     - Y
+   * - bigint
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     - Y
+   * - boolean
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     -
+   * - real
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     -
+   * - double
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+     -
+     -
+   * - varchar
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     - Y
+     -
+   * - timestamp
+     -
+     -
+     -
+     -
+     -
+     -
+     -
+     - Y
+     -
+     - Y
+     -
+   * - date
+     -
+     -
+     -
+     -
+     -
+     -
+     -
+     -
+     - Y
+     -
+     -
+   * - decimal
+     -
+     -
+     -
+     -
+     -
+     -
+     - Y
+     -
+     -
+     -
+     - Y
+
+Examples
+--------
+
+Below are some examples of valid cast expressions.
+
+::
+
+  SELECT cast(12345.12 as bigint); -- 12345
```

Comment:
It might be helpful to organize this by from -> to cast, e.g. have a separate section for each cast with both valid and invalid examples as well as English description of valid inputs.